modules/table/src/test/java/org/apache/ignite/internal/table/distributed/replication/PartitionReplicaListenerTest.java

```diff
@@ -1533,40 +1534,37 @@ void testReadOnlyGetAfterRowRewrite(
 
         txState = null;
 
+        // Delete the same row 2 times within the same transaction to generate garbage rows in storage.
+        // If the data was not preloaded, there will be one deletion actually.
         UUID tx1 = newTxId();
         delete(tx1, br1Pk);
         upsert(tx1, br1);
+        delete(tx1, br1Pk);
 
-        while (true) {
-            delete(tx1, br1Pk);
-
-            if (upsertAfterDelete) {
-                upsert(tx1, br1);
-            }
-
-            Cursor<RowId> cursor = pkStorage().get(br1);
-
-            if (!insertFirst) {
-                if (!upsertAfterDelete) {
-                    assertFalse(cursor.hasNext());
-                }
-
-                // If there were no other entries in index, break after first iteration.
-                break;
-            } else {
-                // This check is only for cases when new rows generation mess the index contents and some rows there have no value.
-                // We try to reach the point when the first row in cursor have no value, to test that this row will be skipped by RO tx.
-                // TODO https://issues.apache.org/jira/browse/IGNITE-18767 after this, the following check may be not needed.
-                RowId rowId = cursor.next();
+        if (upsertAfterDelete) {
+            upsert(tx1, br1);
+        }
 
-                BinaryRow row = testMvPartitionStorage.read(rowId, HybridTimestamp.MAX_VALUE).binaryRow();
+        Cursor<RowId> cursor = pkStorage().get(br1);
```

Comment:
Move the creation of the cursor under the if closure.

Reply:
done